docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+.. Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+   other Shroud Project Developers.
+   See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (BSD-3-Clause)
+
+.. All of the examples are ordered as
+   C or C++ code from users library
+   YAML input
+   C++ wrapper
+   Fortran interface
+   Fortran wrapper
+   Fortran example usage
+   C++ example usage
+
+Sample Fortran Wrappers
+=======================
+
+This chapter gives details of the generated code.
+It's intended for users who want to understand the details
+of how the wrappers are created.
+
+All of these examples are derived from tests in the ``regression``
+directory.
+
+.. _example_NoReturnNoArguments:
+
+No Arguments
+------------
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start NoReturnNoArguments
+   :end-before: end NoReturnNoArguments
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void NoReturnNoArguments()
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start no_return_no_arguments
+   :end-before: end no_return_no_arguments
+   :dedent: 8
+
+If wrapping a C++ library, a function with a C API will be created
+that Fortran can call.
+
+.. literalinclude:: ../regression/reference/tutorial/wrapTutorial.cpp
+   :language: c
+   :start-after: start TUT_no_return_no_arguments
+   :end-before: end TUT_no_return_no_arguments
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    use tutorial_mod
+    call no_return_no_arguments
+
+The C++ usage is similar:
+
+.. code-block:: c++
+
+    #include "tutorial.hpp"
+
+    using namespace tutorial;
+    NoReturnNoArguments();
```

Comment:
Any reason to have ``using namespace tutorial``, instead of just ``tutorial::NoReturnNoArguments()``?

Reply:
No reason. Fixed.